SQLQuery_2.sql

```diff
@@ -0,0 +1,295 @@
+FROM Contracts c
+INNER JOIN Bills b ON c.Id = b.ContractId;
+
+-- 8.Display the contract's information (contract id, start date, end date, contract type, bill id, amount, deadline, isPaid) together with their corresponding bills.
+
+SELECT c.Id 'Contract ID', c.StartDate, c.EndDate, ct.Name, b.Id 'Bill''s ID', b.Amount,b.Deadline, b.IsPaid
+FROM Contracts c
+    LEFT JOIN Bills b ON c.Id = b.ContractId
+    INNER JOIN ContractTypes ct ON ct.Id = c.Id;
+
+-- 9. Display the customers' information (just customer id, email address, contract end date & contract type) together with their corresponding contract type(s).
+
+SELECT cu.Id 'Customer ID', cu.Email, ct.EndDate, ct.ContractType 'ID Contract Type'
+FROM Customers_Contracts cc
+    INNER JOIN Customers cu ON cu.Id = cc.ContractId
+    INNER JOIN Contracts ct ON ct.Id = cc.CustomerId;
+
+-- 10. On exercise above make sure to display even the customers who don't have yet a contract.
+
+SELECT cu.Id 'Customer ID', cu.Email, ct.EndDate, ct.ContractType 'ID Contract Type'
+FROM Customers_Contracts cc
+    INNER JOIN Customers cu ON cu.Id = cc.ContractId
+    RIGHT JOIN Contracts ct ON ct.Id = cc.CustomerId;
+
+-- 11.Display the number of addresses a user has based on his / her email. Tip: use the customer's email when counting and grouping the records.
+
+SELECT COUNT(a.StreetName) 'Number of Addresses'
+FROM Customers c
+INNER JOIN Addresses a ON c.Id = a.CustomerId
+GROUP BY c.Email
+HAVING c.Email = 'am@gmail.com';
+
+-- 12.Create a function that for a specific customer email returns the number of addresses that he / she has registered. The final column name should have the alias 'NumberOfAddreses'.
+
+GO
+CREATE FUNCTION dbo.GetNumberOfAddresses(
+    @inputEmail VARCHAR(100)
+)
+RETURNS INT
+AS
+BEGIN
+    DECLARE @numberOfAddresses INT;
+    SET @numberOfAddresses = (
+        SELECT COUNT(a.StreetName) 'NumberOfAddreses'
+        FROM Customers c
+            INNER JOIN Addresses a ON c.Id = a.CustomerId
+        GROUP BY c.Email
+        HAVING c.Email = @inputEmail)
+    RETURN @numberOfAddresses;
+END;
+GO
+
+SELECT dbo.GetNumberOfAddresses('am@gmail.com') 'NumberOfAddreses';
+
+--  13. Create a view that contains all the information about the customers (id, email) who have unpaid bills, including the bill amount to be paid and the deadline
+
+GO
+CREATE VIEW GetCustomersUnpaidBills
+AS
+    SELECT cc.CustomerId, b.IsPaid, b.Amount, b.Deadline
+    FROM Contracts c
+            INNER JOIN Bills b ON b.ContractId = c.Id
+            INNER JOIN Customers_Contracts cc ON cc.ContractId = c.Id
+    WHERE b.IsPaid = 0;
+GO
+
+SELECT *
+FROM GetCustomersUnpaidBills;
+
+GO
+CREATE VIEW CustomersWithUnpaidBills
+AS
+    SELECT c.Id 'Customer ID', c.Email, gub.Amount 'Amount to be paid', gub.Deadline 
+    FROM GetCustomersUnpaidBills gub
+     INNER JOIN Customers c ON c.Id = gub.CustomerId;
+GO
+
+
+ --  a) Query the view to get all data.
+
+SELECT *
+FROM CustomersWithUnpaidBills;
```

Comment:
Nice that you decided to use two different views. In real life it can lead to a bit of performance loss as there are two views to maintain instead of one but for the learning purposes it's great.